### DISCUSSION TITLE
Option for using Atom for the web-based editor

### DISCUSSION THREAD

**Discussion by unrealapex** [community/community#5184] [Codespaces]
It would be nice to be able to use Atom alongside VS Code in the web based editor <s>because it provides a better UI for working with Git. Some Git features like amending commits aren't available through VS Code's UI(I'd assume for simplicity)</s>.

**Comment by bpasero** [comment]
What about these actions to amend commits?

![image](https://user-images.githubusercontent.com/900690/130562583-0abdb001-f1a4-4df1-9d26-19c783e1b7d5.png)

**Reply by unrealapex** [reply]
I don't have those options in the editor. I'm assuming your screenshot is from Codespaces, what I mean is the web editor with no compute, see [this](https://code.visualstudio.com/docs/remote/codespaces#_browserbased-editor) and also [this](https://docs.github.com/en/codespaces/developing-in-codespaces/web-based-editor).

![Screenshot 2021-08-24 3 53 42 PM](https://user-images.githubusercontent.com/79121360/130700391-f5fb7657-9b3f-4fc8-8cbb-3c3e077ae348.png)

**Comment by metaskills** [comment]
Would this be possible using the new SSH feature and using Atom with a locally mounted SSH drive? Is that even possible?

**Reply by asciimike** [reply]
Following up on this, it feels like something like https://atom.io/packages/remote-atom might be the answer here. Couple that with some new CLI functionality to make `ssh` "just work" (https://github.com/cli/cli/pull/4915) it feels like this might become fairly straightforward in the near future. I think an optimization on top of this might be to make an atom package for codespaces that encapsulates some of this.

**Comment by phaseOne** [comment]
@UnrealApex amending commits isn't shown because the web IDE is effectively pushing changes to the remote GitHub repo on each commit; there is no local git repo with the web IDE. Therefore, [amending a commit would require a force push, and this would break collaboration](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/changing-a-commit-message#amending-older-or-multiple-commit-messages).

Having said that, I too think it would be nice to have the capability to force push changes when working on a feature branch that doesn't have any collaborators. I suggest retitling your post, or split out the request to amend commits as a separate feature. If you don't plan to do this, I will make a new post regarding amending commits.

**Reply by unrealapex** [reply]
Thanks, that explains a lot! New discussion on amending commits [here](https://github.com/github/feedback/discussions/8891#discussion-3744389).

**Comment by jeanp1209** [comment]
>

**Comment by jeanp1209** [comment]
[](url)

**Comment by Abrahamhutal** [comment]
Hello
